BUILD

```diff
@@ -258,22 +258,60 @@ cc_binary(
 ################################################################################
 genrule(
     name = "generate_java_descriptor_proto",
-    tools = [":protoc"],
-    srcs = [ "src/google/protobuf/descriptor.proto", ],
-    outs = [ "com/google/protobuf/DescriptorProtos.java" ],
+    srcs = ["src/google/protobuf/descriptor.proto"],
+    outs = ["com/google/protobuf/DescriptorProtos.java"],
     cmd = "$(location :protoc) --java_out=$(@D)/../../.. $<",
+    tools = [":protoc"],
 )
 
 java_library(
     name = "java_proto",
-    visibility = ["//visibility:public"],
     srcs = glob([
-        "java/src/main/java/com/google/protobuf/*.java"
+        "java/src/main/java/com/google/protobuf/*.java",
     ]) + [
-      ":generate_java_descriptor_proto",
-    ]
+        ":generate_java_descriptor_proto",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+################################################################################
+# Python support
+################################################################################
+
+# A hack that moves Python runtime srcs files form python/google/protobuf/... to
+# bazel-genfiles/google/protobuf/... so that the generated files can correctly
+# import modules from google.protobuf.
+PYTHON_SRCS = glob(
```

Comment:
Maybe better to list these explicitly instead of globbing, to match the rest of the file.

It would also let you omit the leading "python/" so you don't have to strip it away in some places. It's simpler to prepend "python/" when required rather than strip it away when it is not necessary.